### ISSUE TITLE
 Is there any implementation for Sparse Sofmax ?

### ISSUE THREAD

**Issue by sandl99** [rusty1s/pytorch_sparse#138]
Hi,
I want to do Softmax operators for SparseTensor(dim=1). Is this a correct implementation?
```
from torch_scatter import gather_csr, segment_csr
from torch_sparse import SparseTensor

def softmax(src: SparseTensor, dim=1):
    value = src.storage.value()
    rowptr = src.storage.rowptr()

    value_exp = torch.exp(value)
    sum_value_exp = segment_csr(value_exp, rowptr)
    sum_value_exp = gather_csr(sum_value_exp, rowptr)

    return src.set_value(value_exp / sum_value_exp, layout='csr')
```
Thank you!

**Comment by rusty1s** [comment]
Yes, this is correct. We also have a `softmax` implementation here: https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/utils/softmax.py#L36-L38

One thing that is different in this implementation is that you want to first subtract the maximum value of each row in order to guarantee that the `exp` call does not lead to overflows.

**Comment by sandl99** [comment]
Oh, my bad. I should have checked in `pytorch_geometric` lib.
